apps/er-coap/er-coap.c

```diff
@@ -123,7 +123,7 @@ coap_set_option_header(unsigned int delta, size_t length, uint8_t *buffer)
     } else if(*x > 12) {
       buffer[++written] = (*x - 13);
     }
-  } while(x != &length && (x = &length));
+  } while(x != (unsigned int*)&length && (x = (unsigned int*)&length));
 
```

Comment:
At the end of the same line I believe: `(x = (unsigned int*)&length)`.
I agree this piece of code is very obscure... seems to simply run something twice, first with `x` pointing to `delta`, second with `x` pointing to `length`.